apps/src/block_utils.js

```diff
@@ -1324,3 +1325,25 @@ exports.installCustomBlocks = function({
 
   return blocksByCategory;
 };
+
+/**
+ * Adds a second value to options array elements if a second one does not exist.
+ * The second value is used as the generated code for that option.
+ * Also required for backwards compatibility with existing blocks that are missing the second value.
+ *
+ * @param  {string[][]| string[]} dropdownOptions
+ * @returns {string[][]} Sanitized array of dropdownOptions, ensuring that both a first and second value exist
+ */
+const sanitizeOptions = function(dropdownOptions) {
+  const newOptions = [];
+
+  dropdownOptions.forEach(option => {
+    if (option.length === 1) {
+      newOptions.push([option[0], option[0]]);
+    } else {
+      newOptions.push(option);
+    }
+  });
```

Comment:
I think this is a slightly more performant way to handle the logic. Let me know if you think this change makes sense!

Reply:
nit: might be slightly cleaner as a map like `return options.map(option => option.length === 1 ? [option[0], option[0]] : option)`

Reply:
I like doing this in a separate function so that we can have a doc comment for it :)

Reply:
great idea! updated here: https://github.com/code-dot-org/code-dot-org/pull/44313/commits/eed89abfa62e56cd85bf9170e50e5b370707c186